src/dispatch/document/service.py

```diff
@@ -64,10 +66,32 @@ def get_all(*, db_session) -> List[Optional[Document]]:
 
 def create(*, db_session, document_in: DocumentCreate) -> Document:
     """Creates a new document."""
+    # handle the special case of only allowing 1 FAQ document per-project
     project = project_service.get_by_name_or_raise(
         db_session=db_session, project_in=document_in.project
     )
 
+    if document_in.resource_type == DocumentResourceReferenceTypes.faq:
+        faq_doc = (
+            db_session.query(Document)
+            .filter(Document.resource_type == DocumentResourceReferenceTypes.faq)
+            .filter(Document.project_id == project.id)
+            .one_or_none()
+        )
+        if faq_doc:
+            raise ValidationError(
+                [
+                    ErrorWrapper(
+                        ExistsError(
+                            msg="FAQ document already defined for this project.",
+                            document=faq_doc.name,
+                        ),
+                        loc="document",
+                    )
+                ],
+                model=DocumentCreate,
+            )
+
```

Comment:
We should move this to https://github.com/Netflix/dispatch/blob/master/src/dispatch/document/flows.py#L18 when the incident flows code is refactored to use `create_document` in the document/flows.py module.